python/triqs_dft_tools/sumk_dft_tools.py

```diff
@@ -897,7 +897,7 @@ def spaghettis(self, broadening=None, plot_shift=0.0, plot_range=None, ishell=No
 
                 for ish in range(self.shells[ishell]['dim']):
                     for sp in spn:
-                        Akw[sp][ish, ik] = -gf.data[numpy.where((mesh > om_minplot)&(mesh < om_maxplot))].imag.trace(axis1=1, axis2=2)/numpy.pi
+                        Akw[sp][ish, ik] = -G_loc[sp].data[numpy.where((mesh > om_minplot)&(mesh < om_maxplot)),ish,ish].imag/numpy.pi
```

Comment:
Thanks for this fix. How did this ever work before? This is supposed to give an orbital resolved spectral function for `ishell`, where `ish` is an orbital index? If `ish` is an orbital index, can we name it `iorb` or something?

Reply:
In 3.0.x, it was correct (the trace is not needed here). This new version takes advantage of using ``numpy.where`` to quickly slice the mesh window. Yes, ``ish`` is an orbital index. I think ``ish`` is used throughout the code base as the looping index over a ``shell`` dimension, but happy to change to ``iorb``.

Reply:
Okay we keep it as it is. This keeps the behavior and fixes the problem! We should maybe consider at some point specifying that the option `ishell` also returns an orbital resolved Akw.